slang-checks/src/main/java/com/sonarsource/slang/checks/CommentedCodeCheck.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * SonarSource SLang
+ * Copyright (C) 2009-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package com.sonarsource.slang.checks;
+
+import com.sonarsource.slang.api.ASTConverter;
+import com.sonarsource.slang.api.Comment;
+import com.sonarsource.slang.api.TextPointer;
+import com.sonarsource.slang.api.TextRange;
+import com.sonarsource.slang.api.TopLevelTree;
+import com.sonarsource.slang.checks.api.InitContext;
+import com.sonarsource.slang.checks.api.SlangCheck;
+import com.sonarsource.slang.impl.TextRangeImpl;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.sonar.check.Rule;
+
+@Rule(key = "S125")
+public class CommentedCodeCheck implements SlangCheck {
+  private ASTConverter commentAnalyser;
+
+  public CommentedCodeCheck(ASTConverter astConverter) {
```

Comment:
I think you should use another interface for this, as the `ASTConverter` given here modifies the original AST (by wrapping content inside a function), and you actually do not need to get the AST at all here. An interface testing the validity and returning a boolean would be sufficient

Reply:
As discussed, I will add a new method on ASTConverter to handle this logic.